### PR TITLE
Remove IE9 and IE10 workarounds

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by roman-simionov — CHANGES_REQUESTED** [review]
May be pull these changes in v18.2 only to avoid breaks in projects which use v18.1?
//cc @jsdmitry @vconst  @pavelgruba @San4es

**Comment by San4es** [comment]
> May be pull these changes in v18.2 only to avoid breaks in projects which use v18.1?

@roman-simionov 
I do not see any problems to do this in v18.1. We already have [BC](https://www.devexpress.com/bcid=BC4273) and [blog post](https://community.devexpress.com/blogs/aspnet/archive/2018/02/14/ie-9-10-and-old-android-browsers-support-ending-in-devextreme-v18-1.aspx) about it. So, I think we need to do this ASAP to avoid another BC in future major releases.

@dxbykov @AlekseyMartynov @ovchinnikov @pavelgruba 
What do you think?

File: js/client_exporter/image_creator.js [hunk 1]
```diff
@@ -481,9 +481,7 @@ function createFilter(element) {
 
         switch(attr.result.value) {
             case "gaussianBlurResult":
-                if(attr.stdDeviation) { // T511738, IE10
```
Comment (San4es) [inline js/client_exporter/image_creator.js:484]: @roman-simionov 
```if(attr.stdDeviation) {```
Is this condition is actual for IE10 only?

File: js/client_exporter/image_creator.js [hunk 2]
```diff
@@ -541,7 +539,7 @@ function drawCanvasElements(elements, context, parentOptions) {
 function setLineDash(context, options) {
     var matches = options["stroke-dasharray"] && options["stroke-dasharray"].match(/(\d+)/g);
 
-    if(matches && matches.length && context.setLineDash) { // IE10 does not have setLineDash
```
Comment (San4es) [inline js/client_exporter/image_creator.js:544]: @roman-simionov 
>&& context.setLineDash

Are we sure that this part is specific for IE10 only?

File: js/ui/grid_core/ui.grid_core.adaptivity.js [hunk 3]
```diff
@@ -931,19 +930,8 @@ module.exports = {
                     return this.callBase() || !!this._adaptiveColumnsController.getHidingColumnsQueue().length;
                 },
 
-                _updateScrollableForIE: function() {
-                    var that = this;
-
-                    if(browser.msie && parseInt(browser.version) <= 11) {
```
Comment (San4es) [inline js/ui/grid_core/ui.grid_core.adaptivity.js]: >parseInt(browser.version) <= 11

@alexander-kotov-dx 
What about IE11?

**Comment by AlekseyMartynov** [comment]
Ask PM, хуле...

**Review by San4es — CHANGES_REQUESTED** [review]

File: js/client_exporter/image_creator.js (hunk 2 above)
Comment (roman-simionov) [reply js/client_exporter/image_creator.js:544]: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/setLineDash

File: js/ui/grid_core/ui.grid_core.adaptivity.js (hunk 3 above)
Comment (alexander-kotov-dx) [reply js/ui/grid_core/ui.grid_core.adaptivity.js]: looks like my mistake. I'll fix it. Thanks

File: js/client_exporter/image_creator.js (hunk 1 above)
Comment (roman-simionov) [reply js/client_exporter/image_creator.js:484]: I have not found any browser compatibility table for this. We need to check it in all supported browsers

Comment (alexander-kotov-dx) [reply js/client_exporter/image_creator.js:484]: IE 10.0.9200.16384 - IE10 bug
IE 11.0.9600.18098 - it works
IE 11.48.17134.0 - it works
Edge 42.17143.1.0 - it works
Chrome 61, 66 - it works
FF 60.0.1 - it works
Safari 11.1 - it works
Opera 53.0 - it works
Safari (ios 11.4) - no changes
Chrome 66 (Android 8) - it works

File: js/client_exporter/image_creator.js (hunk 2 above)
Comment (San4es) [reply js/client_exporter/image_creator.js:544]: @roman-simionov @alexander-kotov-dx 
What about FF for Android?

Comment (alexander-kotov-dx) [reply js/client_exporter/image_creator.js:544]: android 8, FF 60 - it works

**Comment by dxbykov** [comment]
> What do you think?

I would merge this into the 18.2 branch. I don't think it's a good idea to break something unexpectedly in the very first minor update.

**Comment by San4es** [comment]
>I would merge this into the 18.2 branch. I don't think it's a good idea to break something unexpectedly in the very first minor update.

Ok. Whatever decision we take about 18.1, let's merge this PR in the 18_2 branch **first**. Then we can think about the cherry-pick it to 18_1 additionally. 

@alexander-kotov-dx 
I have changed the base branch to 18_2.

**Review by San4es — APPROVED** [review]

File: styles/mixins.less [hunk 4]
```diff
@@ -42,7 +39,6 @@
     background: -webkit-gradient(radial, center center, 0px, center center, 100%, color-stop(0%, @startColor), color-stop(100%,  @endColor)); /* Chrome,Safari4+ */
     background: -webkit-radial-gradient(center, ellipse cover, @startColor 0%, @endColor 100%); /* Chrome10+, Safari5.1+ */
     background:    -moz-radial-gradient(center, ellipse cover, @startColor 0%, @endColor 100%); /* FF3.6+ */
-    background:     -ms-radial-gradient(center, ellipse cover, @startColor 0%, @endColor 100%); /* IE10+ */
```
Comment (mrjohn0011) [inline styles/mixins.less:45]: This is for IE 10+. What background will be in IE 11 ?

**Review by mrjohn0011 — COMMENTED** [review]
Some css rules with "-ms" prefix have been removed. Does IE11 need "-ms" prefix or not?

**Review by roman-simionov — APPROVED** [review]

File: styles/mixins.less (hunk 4 above)
Comment (alexander-kotov-dx) [reply styles/mixins.less:45]: radial-gradient without prefix works well for IE11. https://caniuse.com/#feat=css-gradients

Comment (mrjohn0011) [reply styles/mixins.less:45]: What about other removed "-ms" prefixes?

**Review by mrjohn0011 — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

File: styles/mixins.less (hunk 4 above)
Comment (alexander-kotov-dx) [reply styles/mixins.less:45]: Removed prefixes do not require for IE11. I have found documentation or wrote an example for every property

**Review by jsdmitry — APPROVED** [review]

**Pull request merged**